### PR TITLE
[plugin.video.odeon] 0.0.8

### PR DESCRIPTION
**Pull request by enderf**
### Description

ODEON (www.odeon.com.ar) is Argentina's new video on demand platform where you'll find the best local content: movies, series, documentaries and short films of the widest variety of genres.

Comedies, dramas, children's content, sports. All together in one place, to enjoy whenever you want and on every device you want, with the best picture and sound quality, speed browsing and video playback.

ODEON allows you to customize your preferences, build your own list of favorites and mantain your browsing history.

ODEON, a window to the stories of Argentina.

ODEON es la nueva plataforma argentina de video a demanda, donde encontrarás el mejor contenido nacional: películas, series, documentales y cortos de la más amplia variedad de géneros.

Comedias, dramas, contenidos infantiles, deportes. Todo junto en un mismo lugar, para que disfrutes cuando quieras y en el dispositivo que quieras, con la mejor calidad de imagen y sonido, velocidad de navegación y reproducción de video.

ODEON te permite personalizar tus preferencias, armar tu propia lista de contenidos favoritos y contar con tu historial de navegación.

ODEON, una ventana a las historias de tu país.

### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a sinlge commit with using the following style: [plugin.video.foo] v1.0.0

Addtional information :
- https://www.odeon.com.ar
- https://arsat.github.io/odeon-kodi/
- Contact: dev@odeon.com.ar

### CODE REVIEW

File: plugin.video.odeon/addon.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="es">ODEON, una ventana a las historias de tu país.</summary>
+        <description lang="es">La plataforma de contenidos audiovisuales de producción nacional, desarrollada en el país. Tus películas y series favoritas, cuando quieras y donde quieras, en la mejor calidad.
+ODEON: El lugar del Cine Argentino.</description>
+        <disclaimer lang="es">Los contenidos audiovisuales de la plataforma ODEON pueden contener escenas no aptas para todo público. Se recomienda la supervisión de un mayor. El uso de ODEON y de este sitio web implica la aceptación de nuestros Términos de uso y de nuestra Política de privacidad. Todos los derechos reservados 2003–2015. República Argentina, Instituto Nacional de Cine y Artes Audiovisuales. Lima 319; Código Postal: C1073AAG, CABA; República Argentina.</disclaimer>
+        <language>es</language>
+        <platform>all</platform>
+        <license>GPLV2</license>
```
Comment (phil65) [inline plugin.video.odeon/addon.xml]: license file says v3? which one is correct?

File: plugin.video.odeon/addon.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon id="plugin.video.odeon"
+       name="Odeon" version="0.0.5"
+       provider-name="ARSAT">
+    <requires>
+        <import addon="xbmc.python" version="2.14.0"/>
```
Comment (phil65) [inline plugin.video.odeon/addon.xml]: this should be 2.24.0

**Comment by enderf** [comment]
Thank you, I'm correcting this. Should be V3.

**Comment by phil65** [comment]
Also, please our translation system and remove the hardcoded strings from your code.

**Comment by enderf** [comment]
new updated version 0.0.7 squashed.
let me know if it's ok now.

**Comment by learningit** [comment]
Actually there are still 2 commits so the squash didn't work. I'll review the 0.0.7

Not sure why you changed the icon.png to 512x512, but it should be 256x256 as you originally had it.

File: plugin.video.odeon/odeon.py [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    token = addon.getSetting('token')
+    url = make_url('{0}/vista/{1}/{2}/{3}/{4}/{5}'.format(source, PID, sid, int(totalTime), int(currentTime), token))
+    r = requests.get(url, headers=get_headers())
+    # xbmc.executebuiltin('Notification(%s, %s, %d, %s)'%(addon_name, 'Beat ' + str(currentTime/60), 1000, addon_icon))
+
+
+def monitor(source, sid, seek=None):
+    # esperamos a que el player comience
+    for i in range(60):
+        if xbmc.Player().isPlayingVideo() or xbmc.abortRequested: break
```
Comment (learningit) [inline plugin.video.odeon/odeon.py]: You should be using xbmc.Monitor.waitForAbort() ( http://mirrors.xbmc.org/docs/python-docs/15.x-isengard/xbmc.html#Monitor-waitForAbort ) possibly here and definitely below for the 3 sec wait.

**Comment by enderf** [comment]
- I squashed 0.0.6 with 0.0.7. Not sure how to do it with 0.0.5.
- http://kodi.wiki/view/Add-on_structure says icons should be 256x256 or 512x512. I asked our designer to develop something "nicer" in 512. Let me know if I must change it back to 256.
- The rest OK, I'll do it.

**Comment by learningit** [comment]
1) You can have a look here on how to do the squash: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit  . If you can't I think we can still do the squash when we merge, but could cause issues for you in future updates, so please try to do the squash.

2) icon.png needs to be 256x256, please fix it.

3) Please squash and resubmit when you make the changes.

**Comment by zag2me** [comment]
I just updated the wiki about the 256x256 icon. I also tried 512x512 and it doesn't scale well for some reason inside Kodi. Also uses a lot of bandwidth on the mirrors.

The idea was to look nice on 4K screens, but we are not at that point yet so 256 is the way to go.

**Comment by learningit** [comment]
Looks good to go. I'm going to merge. One question and a couple of observations.
I see that you are writing a value called 'metadata' into the add-on settings. How large is the data being written? If it is large, you may want to consider using a file instead.

I also see that you are using gzip to unzip what I think is a compressed web request.  Given that you're running the webserver in this situation I'm not quite sure what's going on with the compression. You may find zlib less complex. You can look at a getRequest routine that I wrote to see how i use it: https://github.com/xbmc/repo-scripts/blob/isengard/script.module.t1mlib/lib/t1mlib.py#L61

**Comment by enderf** [comment]
'metadata' has now 424 bytes that almost never change. Since this request
has no max-age, it will not be cached. I'll talk to the backend programmer
to fix it.
In my tests gzip worked fine. I'll look into zlib and try it.
Thank you very much.

2016-11-11 19:03 GMT-03:00 learningit <notifications@github.com>:

**Pull request merged**